### ISSUE TITLE
[PRE REVIEW]: Atlas toolkit for Python

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1369]
**Submitting author:** @epeios-q37 (<a href="http://orcid.org/0000-0002-5928-236X">Claude Simon</a>)
**Repository:** <a href="http://github.com/epeios-q37/atlas-python" target ="_blank">http://github.com/epeios-q37/atlas-python</a>
**Version:** 0.7
**Editor:** Pending
**Reviewer:** Pending

**Author instructions**

Thanks for submitting your paper to JOSS @epeios-q37. **Currently, there isn't an JOSS editor assigned** to your paper.

@epeios-q37 if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for an editor to be assigned and for them to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1369 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/lib/whedon.rb:83:in `check_fields': Paper YAML header is missing expected fields: bibliography (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/lib/whedon.rb:69:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/bin/whedon:55:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-a1723d160bb6/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by kyleniemeyer** [comment]
Hi @epeios-q37, before we proceed with finding an editor for your submission, could you provide some justification as to how your software is research software? We have these overall requirements (among others):
 - The software should have an obvious research application
- The software should be a significant contribution to the available open source software that either enables some new research challenges to be addressed or makes addressing research challenges significantly better (e.g., faster, easier, simpler)

Assisting users in creating GUIs is useful, but it isn't clear to me that it is solving a research need. (Meaning, this is useful software, but I don't think it is research software.)

**Comment by epeios-q37** [comment]
Hi @kyleniemeyer. Thank you for handling my submission.

Most of the research software have a command-line interface. With such an interface, a software is very handy, but not easy to use for someone who is not familiar with it, especially if the software has numerous parameters, which are set through command line arguments and/or configuration files. You often have to go deep into the documentation to find the right parameters and their correct values for the software to operate as wanted.

With a software with a graphical user interface, on the contrary, you need less the documentation, and often you don't need the documentation at all, as the interface is self-explanatory. The widgets dedicated to each parameter have usually a tooltip explaining its meaning, and the values it accepts. If needed, there may also be a link that opens the documentation directly to the part dedicated to this parameter.

The software, with a CLI only, can also be provided with an independent companion software, this one with a GUI, which can be used to generate the appropriate command-line parameters and/or configuration files, corresponding to what the user has input.

The GUI has many advantages over a CLI, although the CLI is irreplaceable in some cases. The best is that the software have both a CLI and a GUI, so the user can choose the interface he wants to use. For example, the GUI to become familiar with the software, and the CLI when he wants to use the software in a script.

But whoever writes research software wants to focus on the algorithms he implements in it. He cannot spend much time developing the GUI. And he has absolutely no time to learn a dedicated language and/or to master a GUI framework and all its component. Not to mention that the use of such a framework will make the installation of its software very cumbersome.

The *Atlas* toolkit avoids this. It's very lightweight, available in *Python*, a language very popular in the scientific community (note that the *Atlas* toolkit is also available for *Java*, *Node.js*, *PHP* and *Ruby*, and will be also available for other languages in the future), and requires only basic knowledge of HTML, which is easy to learn, and for which you can easily find help if needed. Actually, most developers already know HTML.

With the *Atlas* toolkit, you can easily add a GUI to a software, even if this software wasn't originally meant to have a GUI. Thus, the software will have both interfaces. When the software is launched in GUI mode, a web browser opens automatically, displaying the interface of the software. And, if you want to show the software to other people, you can give them access to the one installed on your computer, so they don't have to install it on their own computer, simply by giving them the URL opened in the web browser, which is also automatically displayed as QR code. You don't have to configure the router to which your computer is connected, nor do you have to deploy the software on a remote server.

To summarize (sorry, it was a bit long), the *Atlas* toolkit is meant as a way for someone who writes research software to write them more easy to use. So, I thought that the *Atlas* toolkit met your criteria. Maybe did I misunderstood something, as English is not my first language (I'm French)...

**Comment by kyleniemeyer** [comment]
Hi @epeios-q37, sorry for the delay. The editorial board was deciding whether to consider your software as in scope for JOSS or not. 

Unfortunately, after some discussion, we decided that while your software does look useful, it is fundamentally a utility and not research software itself, and thus not in JOSS's scope for submissions.
